api/hypershift/v1beta1/hostedcluster_types.go

```diff
@@ -2008,8 +2008,118 @@ type AzurePlatformSpec struct {
 	// +kubebuilder:validation:XValidation:rule="self == oldSelf",message="SecurityGroupID is immutable"
 	// +kubebuilder:validation:Required
 	// +immutable
-	// +required
-	SecurityGroupID string `json:"securityGroupID,omitempty"`
+	SecurityGroupID string `json:"securityGroupID"`
+
+	// keyVaultManagedIdentityClientID is the client ID of the managed identity which can access the Azure Key Vaults,
+	// in the management cluster, to retrieve secrets and certificates, like certificates related to the
+	// ControlPlaneManagedIdentities for example. The HostedCluster controller populates this field for ARO HCP
```

Comment:
> The HostedCluster controller populates this field for ARO HCP
	// deployments if it isn't already initialized.
	//

Can we drop this statement? I don't think we want the controller to populate its own API. We can just let it default transparently without updating the field if we want to.

Reply:
Discussed this on Slack with Alberto:

CS is going to use the same MI here for all HCs on a particular mgmt cluster. They wanted to pass this value in through the HO installation ([I'm setting it as an env var on the HO pod here](https://github.com/openshift/hypershift/pull/4877/commits/43274102ccf8e3f906534e26481cef9ae563a6da#diff-4edc0a874ce10f9d4f77594c9ff489c3275cb2adc788826a9ea2f48a64c336efR486)) and not populate it on each HC CR. We need this value on the CPO side for creating the SecretProviderClass CRs.

This field should get set in the HCP, at its creation time.

Reply:
Should this be part of status then instead of spec?

Reply:
@bennerv @miguelsorianod - I've updated this now. The field here is populated when the HCP is created.

Reply:
I think we didn't understand each other when discussing this.

We understood the attribute would be populated in the HCP and not in the HC itself. The original comment by Ben still applies then

Reply:
Update - we discussed this offline (Miguel, Ben, & myself). We are going to remove this field and the HO will pass the KY MI client ID through an env var to the CPO.